vmm_tests/vmm_tests/tests/tests/x86_64/openhcl_linux_direct.rs

```diff
@@ -155,8 +155,112 @@ fn new_test_vtl2_nvme_device(
+                    })
+                    .collect::<Vec<_>>();
+
+                c.vpci_devices.extend(
+                    device_ids
+                        .iter()
+                        .map(|(nsid, guid)| new_test_vtl2_nvme_device(*nsid, SIZE, *guid, None)),
+                );
+            })
+            .with_custom_vtl2_settings(|v| {
+                let device_ids = (0..NUM_NVME_DEVICES)
+                    .map(|i| {
+                        let mut g = BASE_GUID;
+                        g.data2 = g.data2.wrapping_add(i as u16) + 0x1110;
+                        (NSID_OFFSET + i as u32, g)
+                    })
+                    .collect::<Vec<_>>();
+
+                v.dynamic.as_mut().unwrap().storage_controllers.push(
+                    vtl2_settings_proto::StorageController {
+                        instance_id: Guid::new_random().to_string(),
+                        protocol: vtl2_settings_proto::storage_controller::StorageProtocol::Scsi
+                            .into(),
+                        luns: device_ids
+                            .iter()
+                            .map(|(nsid, guid)| vtl2_settings_proto::Lun {
+                                location: (*nsid - NSID_OFFSET) + 1, // Add 1 so as to avoid any confusion with booting from LUN 0 (on the implicit SCSI controller created by the above `config.with_vmbus_redirect` call above).
+                                device_id: Guid::new_random().to_string(),
+                                vendor_id: "OpenVMM".to_string(),
+                                product_id: "Disk".to_string(),
+                                product_revision_level: "1.0".to_string(),
+                                serial_number: "0".to_string(),
+                                model_number: "1".to_string(),
+                                physical_devices: Some(vtl2_settings_proto::PhysicalDevices {
+                                    r#type:
+                                        vtl2_settings_proto::physical_devices::BackingType::Single
+                                            .into(),
+                                    device: Some(vtl2_settings_proto::PhysicalDevice {
+                                        device_type:
+                                            vtl2_settings_proto::physical_device::DeviceType::Nvme
+                                                .into(),
+                                        device_path: guid.to_string(),
+                                        sub_device_path: *nsid,
+                                    }),
+                                    devices: Vec::new(),
+                                }),
+                                ..Default::default()
+                            })
+                            .collect(),
+                        io_queue_depth: None,
+                    },
+                )
+            })
+        })
+        .run()
+        .await?;
```

Comment:
This continues to feel awkward / too much code to do a simple thing like add disks to vtl2.